app/assets/javascripts/discourse/app/components/modal/history.js

```diff
@@ -0,0 +1,337 @@
+import Component from "@glimmer/component";
+import I18n from "I18n";
+import Post from "discourse/models/post";
+import { categoryBadgeHTML } from "discourse/helpers/category-link";
+import { iconHTML } from "discourse-common/lib/icon-library";
+import { sanitizeAsync } from "discourse/lib/text";
+import { inject as service } from "@ember/service";
+
+function customTagArray(val) {
+  if (!val) {
+    return [];
+  }
+  if (!Array.isArray(val)) {
+    val = [val];
+  }
+  return val;
+}
+
+export default class History extends Component {
+  @service dialog;
+  @service site;
+  @service currentUser;
+  @service siteSettings;
+
+  @tracked loading;
+  @tracked postRevision;
+  @tracked viewMode = this.site?.mobileView ? "inline" : "side_by_side";
+
+  constructor() {
+    super(...arguments);
+    this.refresh(this.args.model.postId, this.args.model.postVersion);
+  }
+
+  get loadFirstDisabled() {
+    return (
+      this.loading ||
+      this.postRevision?.current_revision <= this.postRevision?.first_revision
+    );
+  }
+
+  get loadPreviousDisabled() {
+    return (
+      this.loading ||
+      (!this.postRevision.previous_revision &&
+        this.postRevision.current_revision <=
+          this.postRevision.previous_revisionI)
+    );
+  }
+
+  get loadNextDisabled() {
+    return (
+      this.loading ||
+      this.postRevision?.current_revision >= this.postRevision?.next_revision
+    );
+  }
+
+  get loadLastDisabled() {
+    return (
+      this.loading ||
+      this.postRevision?.current_revision >= this.postRevision?.next_revision
+    );
+  }
+
+  get displayRevisions() {
+    return this.postRevision?.version_count > 2;
+  }
+
+  get modalTitleKey() {
+    return this.args.model.post.version > 100
+      ? "history_capped_revisions"
+      : "history";
+  }
+
+  get previousVersion() {
+    return this.postRevision?.current_version - 1;
+  }
+
+  get revisionsText() {
+    return I18n.t(
+      "post.revisions.controls.comparing_previous_to_current_out_of_total",
+      {
+        previous: this.previousVersion,
+        icon: iconHTML("arrows-alt-h"),
+        current: this.postRevision?.current_version,
+        total: this.postRevision?.version_count,
+      }
+    );
+  }
+
+  get titleDiff() {
+    let mode = this.viewMode;
+    if (mode === "side_by_side_markdown") {
+      mode = "side_by_side";
+    }
+    return this.postRevision?.title_changes?.[mode];
+  }
+
+  get bodyDiff() {
+    let html = this.postRevision?.body_changes?.[this.viewMode];
+    if (this.viewMode === "side_by_side_markdown") {
+      return html;
+    } else {
+      const opts = {
+        features: { editHistory: true, historyOneboxes: true },
+        allowListed: {
+          editHistory: { custom: (tag, attr) => attr === "class" },
+          historyOneboxes: ["header", "article", "div[style]"],
+        },
+      };
+      sanitizeAsync(html, opts).then((result) => (html = result));
+      return html;
```

Comment:
This is kinda weird... we kick off some async process, but then we immediately return the value of `html`. I don't think the result of the async operation is ever actually being used? 🤔